pkg/spdk/server.go

```diff
@@ -114,6 +125,21 @@ func (s *Server) monitoring() {
 	}
 }
 
+func (s *Server) tryEnsureSPDKTgtHealthy() error {
+	running, err := util.IsSPDKTargetProcessRunning()
+	if err != nil {
+		return errors.Wrap(err, "failed to check spdk_tgt is running")
+	}
+
+	if running {
+		logrus.Info("SPDK Server: reconnecting to spdk_tgt")
+		return s.spdkClient.Reconnect()
+	}
+
+	logrus.Info("SPDK Server: restarting spdk_tgt")
+	return util.StartSPDKTgtDaemon()
```

Comment:
Does this mean that the instance manager (daemon) is responsible for the first time spdk_tgt start but the spdk engine server will take care of the following spdk_tgt restart? This looks a little bit weird...

Reply:
Yes.
But longhorn-engine handles/restarts `tgtd` using the same weird pattern... Let's improve them in the future.
https://github.com/longhorn/longhorn-engine/blob/master/vendor/github.com/longhorn/go-iscsi-helper/iscsidev/iscsi.go#L88

Reply:
Let's make it to the same place in a later version.